.github/workflows/vuln-check-reusable.yaml

```diff
@@ -0,0 +1,151 @@
+
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+
+    steps:
+      - name: Checkout vuln-check scripts
+        uses: actions/checkout@v4
+        with:
+          repository: scalar-labs/actions
+          sparse-checkout: vuln-check-script
+          ref: main
+          path: vuln-check
+
+      - id: prepare-target-ref
+        name: Prepare target-ref
+        env:
+          SCRIPT_PATH: ./vuln-check/vuln-check-script
+        run: |
+          # Find the latest release name if `find-latest-release` is set to true. Use the value of `target-ref` as is otherwise.
+          if [[ ${{ inputs.find-latest-release }} = 'true' ]]; then
+            releases=$($SCRIPT_PATH/fetch_gh_releases "${{ github.event.repository.owner.login }}" "${{ github.event.repository.name }}")
+            echo -------------
+            echo "releases: $releases"
+            echo -------------
+            target_release=''
+            if [[ -n $releases ]]; then
+              target_release=$($SCRIPT_PATH/find_latest_release "${{ inputs.target-ref }}" $releases)
+            fi
+            if [[ -z $target_release ]]; then
+              echo "Can't find a target release"
+              exit 1
+            fi
+            echo $target_release
+            echo "target-ref=$target_release" >> $GITHUB_OUTPUT
+          else
+            echo "target-ref=${{ inputs.target-ref }}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Checkout the target repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ steps.prepare-target-ref.outputs.target-ref }}
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v4
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+
+      - name: Docker build
+        uses: gradle/gradle-build-action@v3
+        with:
+          arguments: docker
+
+      - name: Set version
+        id: version
+        run: |
+          VERSION=$(${{ inputs.version-command }})
+          echo "version=${VERSION}" >> $GITHUB_OUTPUT
+
+  vuln-check:
+    runs-on: ubuntu-latest
+
+    needs: build
+
+    strategy:
+      matrix:
+        image: ${{ fromJSON(inputs.images) }}
+      fail-fast: false
+      max-parallel: 1
+
+    continue-on-error: true
+
+    steps:
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+
+      - name: Run Trivy vulnerability scanner for ${{ matrix.image[0] }}
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: ghcr.io/scalar-labs/${{ matrix.image[1] }}:${{ needs.build.outputs.version }}
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          severity: 'CRITICAL,HIGH'
+          timeout: '60m'
+
+      - name: Post Trivy vulnerability check failure for ${{ matrix.image[0] }}
```

Comment:
Can we parameterize whether to post the failure notification to Slack?

We have two workflows: `Scheduled Vulnerability Check` and `Manual Vulnerability Check Trigger`, and I don’t think it’s necessary to post failure notifications to Slack for the `Manual Vulnerability Check Trigger`. It would be helpful if we could toggle whether to post the failure notification to Slack or not.

Reply:
@brfrn169 Thank you for the comment. I made the Slack webhook URL secret not required. By not specifying the URL in the Manual Vulnerability Check Trigger, the workflow will not post a message. Also, I added descriptions of the secrets.

Reply:
I updated the caller side in https://github.com/scalar-labs/scalardb/pull/2288/commits/4123c73b6ac3cefffb976e25a79cf16ff2f29dd6 .  If there are no problems with this secret change, I will make the same modifications to the `scalar` and `scalardb-cluster` repositories as well.